### PR TITLE
add optional `length` argument to scan

### PR DESCRIPTION
**Pull request by mattjj**
cc @shoyer

### CODE REVIEW

**Comment by shoyer** [comment]
Could you add my pseudocode example from https://github.com/google/jax/pull/1677#issuecomment-554685189 to the docstring?

Also: what about setting the default value `xs=None`?

**Comment by mattjj** [comment]
Great idea to update the dosctring; done! I'm not sold on making `xs=None` a default, though, because I don't like default argument values in general.

**Comment by shoyer** [comment]
Fine to skip the default value -- I agree that this is a bit of an edge case so setting a default is a little strange. But do update the default value in the docstring psuedocode to match :)

**Comment by shoyer** [comment]
Thanks Matt!

File: jax/lax/lax_control_flow.py [hunk 1]
```diff
@@ -493,6 +495,9 @@ def scan(f, init, xs):
     xs: the value of type ``[a]`` over which to scan along the leading axis,
       where ``[a]`` can be an array or any pytree (nested Python
       tuple/list/dict) thereof with consistent leading axis sizes.
+    length: optional integer specifying the number of loop iterations, which
+      must agree with the sizes of leading axes of the arrays in ``xs`` (but can
+      be used to perform scans where no input ``xs`` are needed).
```
Comment (gnecula) [inline jax/lax/lax_control_flow.py]: Is it true that if no xs are provided then the second argument to `f` will be `None`?

Comment (mattjj) [reply jax/lax/lax_control_flow.py]: Good question! I believe that if `xs` is an empty container, then the second argument to `f` will be the same empty container. That is, if `xs=None` then the second argument to `f` is `None`, but if `xs=()` then the second argument to `f` is `()`, and if `xs=[()]` then the second argument to `f` is `[()]`.

**Pull request merged**